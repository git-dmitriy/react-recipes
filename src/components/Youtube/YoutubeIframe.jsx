export function YoutubeIframe({ address }) {
  return (
    <div className='relative h-0 mt-8' style={{ paddingBottom: '56.25%' }}>
      <iframe
        className='absolute top-0 left-0 w-full h-full'
        title='Video instruction'
        width='100%'
        height='500px'
        src={`https://www.youtube.com/embed/${address}`}
        allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
        allowFullScreen
      ></iframe>
    </div>
  );
}
