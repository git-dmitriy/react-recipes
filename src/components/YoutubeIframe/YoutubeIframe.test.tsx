import renderer from 'react-test-renderer';
import { YoutubeIframe } from './YoutubeIframe';

it('should render the iframe element with current link', () => {
  const link = 'FXjYU2Ensck';

  const tree = renderer.create(<YoutubeIframe address={link} />).toJSON();

  expect(tree).toMatchSnapshot();
});
