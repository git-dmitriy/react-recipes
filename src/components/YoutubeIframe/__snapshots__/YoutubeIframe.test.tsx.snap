// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the iframe element with current link 1`] = `
<div
  className="relative h-0 mt-8"
  style={
    Object {
      "paddingBottom": "56.25%",
    }
  }
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    className="absolute top-0 left-0 w-full h-full"
    height="500px"
    src="https://www.youtube.com/embed/FXjYU2Ensck"
    title="Video instruction"
    width="100%"
  />
</div>
`;
